cachi2/core/package_managers/bundler/main.py

```diff
@@ -24,5 +34,67 @@ def fetch_bundler_source(request: Request) -> RequestOutput:
 
 def _resolve_bundler_package(package_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
     """Process a request for a single bundler package."""
-    parse_lockfile(package_dir)
+    dependencies = parse_lockfile(package_dir)
+    _get_main_package_name_and_version(package_dir, dependencies)
     return []
+
+
+def _get_main_package_name_and_version(
+    package_dir: RootedPath,
+    dependencies: ParseResult,
+) -> tuple[str, Optional[str]]:
+    """
+    Get main package name and version.
+
+    The main package is the package that is being processed by cachi2.
+    Not any of its dependencies.
+    """
+    path_dependencies = [dep for dep in dependencies if isinstance(dep, PathDependency)]
+    name_and_version = _get_name_and_version_from_path_deps(path_dependencies)
+    if name_and_version is not None:
+        return name_and_version
+
+    log.info("Failed to infer name and version from gemspec file")
+
+    # fallback to origin remote
+    try:
+        name = _get_repo_name_from_origin_remote(package_dir)
+    # if the package directory is not a git repository
+    except UnsupportedFeature:
```

Comment:
This is probably something that already happens regardless of the current PR here, but I was able to get a `InvalidGitRepositoryError` trying to process a folder that is not a repo. I think this is indeed a case for failure, but the error was not treated. Should we fix this as part of this PR?

Reply:
Yeah, good point; this is not handled in `get_repo_id.` I should have added a test for that.

Reply:
I added a new test for this case, that is currently being skipped. I don't know, but it might be considered an empty stub or dead code until it is fixed in `get_repo_id`.

Reply:
So we'll fix it as a follow up?

Reply:
Done.